phac-nml/irida-next
# Data Table Bot Accounts: Remove PAT panel on destroy

## What does this PR do and why?
This PR is a followup to [#849](https://github.com/phac-nml/irida-next/pull/849), and adds logic to remove the new PAT frame upon bot deletion or revoked PAT. This is because we wouldn't want the new PAT panel to persist upon the potential removal of the PAT itself or its associated bot. Due to the changes in response logic in #849, the PAT frame now uses `data-turbo-permanent` attribute to persist through response refreshes, and requires explicit logic to remove/hide it.

## Screenshots or screen recordings
Before:
[Screencast from 2024-12-19 01:17:14 PM.webm](https://github.com/user-attachments/assets/2c2be1e5-9343-4834-bfab-5676c7b36e05)

Now:
[Screencast from 2024-12-19 01:17:53 PM.webm](https://github.com/user-attachments/assets/76816d1b-f443-4c6d-9107-314c34d4bdfa)

## How to set up and validate locally
Repeat for both project and group bots:
1. Navigate to the bot accounts page
2. Create a new bot or generate a new token for an existing bot. Afterwards, the new PAT frame should render.
3. Delete a bot. The new PAT frame should no longer appear.
4. Repeat step 2, and afterwards, revoke a PAT from a bot. Once again, the new PAT frame should no longer appear.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
